File: client/java/src/com/atteq/asl/tasks/TaskResult.java
package com.atteq.asl.tasks;

import com.atteq.asl.results.GenericResult;

public class TaskResult<T> extends GenericResult<T> {

	public TaskResult(Task task, T result) {
		super(task, result);
	}

	public Task getTask() {
		return getPerformer();
	}

	@Override
	public Task getPerformer() {
		return (Task) super.getPerformer();
	}

}

File: client/php/asl/task_result.php
<?php

namespace AtteqServiceLayer;

abstract class TaskResult {
	
	abstract function get_task();
	abstract function get_result();

}

class RawTaskResult extends TaskResult {

	protected $_task;
	protected $_result;

	function __construct(Task $task, $result) {
		$this->_task = $task;
		$this->_result = $result;
	}

	function get_task() {
		return $this->_task;
	}

	function get_result() {
		return $this->_result;
	}
}
